src/main/serialportMap.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-4-Clause
+ */
+
+import { AutoDetectTypes } from '@serialport/bindings-cpp';
+import { WebContents } from 'electron';
+import { SerialPort, SerialPortOpenOptions } from 'serialport';
+
+export type Renderer = Pick<WebContents, 'on' | 'send' | 'id'>;
+export type OpenPort = {
+    serialPort: SerialPort;
+    renderers: Renderer[];
+    settingsLocked: boolean;
+    opening: boolean;
+    options: SerialPortOpenOptions<AutoDetectTypes>;
+};
+
+const mapPath = (path: string) =>
+    process.platform === 'win32' ? path.toLowerCase() : path;
+
+class SerialPortMap {
+    map = new Map<string, OpenPort>();
+
+    set = (path: string, value: OpenPort) => {
+        this.map.set(mapPath(path), value);
+    };
+
+    get = (path: string) => this.map.get(mapPath(path));
+
+    has = (path: string) => this.map.has(mapPath(path));
+
+    delete = (path: string) => this.map.delete(mapPath(path));
+
+    clear = () => {
+        this.map.clear();
+    };
+
+    broadCast = (path: string, channel: string, ...args: unknown[]) => {
+        this.get(path)?.renderers.forEach(renderer => {
+            renderer.send(channel, ...args);
+        });
+    };
+
+    hasEqualOptions = (
+        path: string,
+        options: SerialPortOpenOptions<AutoDetectTypes>
+    ) => {
+        const oldOptions = this.get(path)?.options;
+        if (
+            !oldOptions ||
+            Object.keys(options).length !== Object.keys(oldOptions).length
+        ) {
+            return false;
+        }
+
+        return Object.entries(options).every(([key, value]) => {
+            if (
```

Comment:
Could you maybe elaborate with an example? Do you mean that we do not have to check if the lengths are equal?

Reply:
We might also refactor this code eventually but for now it is mostly a verbatim copy of the code that existed previously: https://github.com/NordicSemiconductor/pc-nrfconnect-launcher/blob/2906233fa5f0ec390953424df640c10cccc15785/src/main/serialport.ts#L58-L75

If I understand you correctly, you consider that we could ignore `undefined` properties? Seems wrong to me: E.g. looking at the property `parity`. If it is `undefined` then it defaults to `false`, according to https://serialport.io/docs/api-bindings-cpp#open. If now someone specifies it as `true` then this is a different setting, so that should not be returned as “equal options”.